LeetCodeAssistant/LeetCodeProblemDetail.swift

```diff
@@ -0,0 +1,5 @@
+import Foundation
+
+protocol LeetCodeProblemDetail {
```

Comment:
- It should be `LeetCodeProblemDescription` or `LeetCodeProblemContent`. Because even it is named as "detail", there's nothing meaning so
- It should be type alias, not protocol, struct nor class